src/edu/csus/ecs/pc2/clics/API202306/ScoreboardService.java

```diff
@@ -23,69 +25,78 @@
 import edu.csus.ecs.pc2.core.log.Log;
 import edu.csus.ecs.pc2.core.model.ContestTime;
 import edu.csus.ecs.pc2.core.model.IInternalContest;
-import edu.csus.ecs.pc2.services.core.ScoreboardJson;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+import edu.csus.ecs.pc2.services.eventFeed.WebServer;
 /**
  * Webservice to handle scoreboard requests
  * 
  * @author ICPC
  */
-@Path("/contest/scoreboard")
+@Path("/contests/{contestId}/scoreboard")
 @Produces(MediaType.APPLICATION_JSON)
 @Provider
 @Singleton
 public class ScoreboardService implements Feature {
 
-    private IInternalContest contest;
+    private IInternalContest model;
     private IInternalController controller;
 
     public ScoreboardService(IInternalContest inContest, IInternalController inController) {
         super();
-        this.contest = inContest;
+        this.model = inContest;
         this.controller = inController;
     }
 
     /**
      * This method returns a representation of the current contest scoreboard in JSON format. 
-     * The return JSON is in the format defined by {@link ScoreboardJson#createJSON(IInternalContest)}.
+     * The return JSON is in the format defined by the CLICS 2023-06 spec.
      * 
-     * @return a {@link Response} object containing the JSON scoreboard
+     * @param servletRequest
+     * @param sc
+     * @param contestId
+     * @param group_id - optional group id query param
+     * @return {@link Response} object containing the JSON scoreboard
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public Response getScoreboard(@Context HttpServletRequest servletRequest, @Context SecurityContext sc) {
-
-        String jsonScoreboard;
-        try {
-            ContestTime contestTime = contest.getContestTime();
-            // verify contest has started or user is an admin
-            if (contestTime.getElapsedMS() > 0 || sc.isUserInRole("admin")) {
-                //ok to return scoreboard
+    public Response getScoreboard(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, @QueryParam("group_id") String group_id) {
 
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
```

Comment:
I think this was because we didn't set a contest Indentifier so it was hash.

Reply:
Correct.  I suppose this will be fixed when we get that info from the contest.json/contest.yaml.

Reply:
when will that be done?

Reply:
When the newer format CDP or CPF is used to configure PC2.  But before we do that, I have to rebase this PR onto the latest **develop** branch, and there will be lots of conflicts - since I added all that group stuff it will certainly confuse git - files moved and changed (event feed stuff for the older protocol revision).